posthog/year_in_posthog/crypto.py

```diff
@@ -0,0 +1,41 @@
+from base64 import urlsafe_b64decode as b64d
+from base64 import urlsafe_b64encode as b64e
+from datetime import timedelta
+
+from cryptography.fernet import Fernet
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.primitives import hashes
+from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
+
+from posthog.cache_utils import cache_for
+from posthog.settings import SECRET_KEY
+
+SECRET_BYTES = SECRET_KEY.encode()
+ITERATIONS = 100_000
+
+backend = default_backend()
+
+"""
+All based on this awesome StackOverflow answer: https://stackoverflow.com/a/55147077
+"""
+
+
+@cache_for(timedelta(seconds=3600))
```

Comment:
This cache only applies for a single web pod - will this even work?

Reply:
Hmmm, good q. I'm not 100% and this definitely falls into the too much for first test in prod...